app/channels/synchronized_editor_channel.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class SynchronizedEditorChannel < ApplicationCable::Channel
+  def subscribed
+    stream_from specific_channel
+  end
+
+  def unsubscribed
+    # Any cleanup needed when channel is unsubscribed
+    stop_all_streams
+  end
+
+  def specific_channel
+    reject unless ProgrammingGroupPolicy.new(current_user, programming_group).stream_sync_editor?
+    "synchronized_editor_channel_group_#{programming_group.id}"
+  end
+
+  def programming_group
+    @programming_group ||= current_user.programming_groups.find_by(exercise_id: params[:exercise_id])
```

Comment:
I cannot access the session at this position. Maybe because the `SynchronziedEditorChannel` only inherits from `ApplicationCable::Channel`. Adding `include ActionController` and `include ApplicationHelper` also didn't help and is maybe a bit overkill when we only want to get the session data.